### DISCUSSION TITLE
next.js on cloudfront + s3 static hosting redirects all pages to homepage

### DISCUSSION THREAD

**Discussion by TulyOpt** [vercel/next.js#28506] [Help]
Hello, 

I am trying to setup a nextjs app using cloudfront and an s3 static hostingMy problem is with the redirection : 
* when I paste the url (https://xxx.cloudfront.net/ ) it works fine, i can click on "Go to pages/post/[pid].js" link and i am well redirected to the 'post'page.
* the problem is when I paste directly "https://xxx.cloudfront.net/post/abc " : The homepage is displayed with the "https://xxx.cloudfront.net/post/abc " url in the browser. next js does not do internal redirection

Do you have an idea of what is wrong ?

Many thanks !

**Comment by ingdaeho** [comment]
did you solved problem??
